exporter/src/test/scala/uk/gov/nationalarchives/consignmentexport/KeycloakClientSpec.scala

```diff
@@ -0,0 +1,58 @@
+package uk.gov.nationalarchives.consignmentexport
+
+import cats.effect.{ContextShift, IO}
+import org.keycloak.admin.client.Keycloak
+import org.keycloak.admin.client.resource.{RealmResource, UserResource, UsersResource}
+import org.keycloak.representations.idm.UserRepresentation
+import sttp.client.{HttpURLConnectionBackend, Identity, NothingT, SttpBackend}
+import uk.gov.nationalarchives.consignmentexport.Config.{Api, Auth, Configuration, EFS, S3}
+
+import scala.concurrent.ExecutionContextExecutor
+
+class KeycloakClientSpec extends ExportSpec {
+  implicit val backend: SttpBackend[Identity, Nothing, NothingT] = HttpURLConnectionBackend()
+  implicit val executionContext: ExecutionContextExecutor = scala.concurrent.ExecutionContext.global
+  implicit val contextShift: ContextShift[IO] = IO.contextShift(executionContext)
+
+  "the getUserDetails method" should "return the correct user details" in {
+    val config = Configuration(S3("", "", ""), Api(""), Auth("authUrl", "clientId", "clientSecret", "realm"), EFS(""))
+    val mockKeycloak = mock[Keycloak]
+    val mockRealmResource = mock[RealmResource]
+    val mockUsersResource = mock[UsersResource]
+    val mockUserResource = mock[UserResource]
```

Comment:
How did you find writing these tests? Do they seem like they'll catch real problems? Given the number of external resources that we have to mock here, would the tests be more reliable and easier to follow if we used WireMock with some HTTP response fixtures? That's my intuition, but you're more familiar with the Keycloak client library than me.